pipeline/matcher/src/test/scala/uk/ac/wellcome/platform/matcher/fixtures/MatcherFixtures.scala

```diff
@@ -132,8 +132,8 @@ trait MatcherFixtures
     sendNotificationToSQS(queue, body = links.workId)
   }
 
-  def ciHash(str: String): String =
-    DigestUtils.sha256Hex(str)
+  def ciHash(ids: String*): String =
+    DigestUtils.sha256Hex(ids.sorted.mkString("+"))
```

Comment:
I know this PR is just generalising this, but why do we do this?

Reply:
This is one of the kludges we do for the matcher implementation.

Quick matcher recap: it looks at all the MergeCandidates on Works, and uses those to construct a graph of all the Works. Works in the same graph component can potentially be merged together.

<img width="768" alt="Screenshot 2021-03-23 at 16 05 24" src="https://user-images.githubusercontent.com/301220/112178567-c2696b80-8bf1-11eb-8ea5-4409428c0ed9.png">

We store these graphs in DynamoDB, with one Dynamo item per Work / graph node.

When you update one Work, you want to find all the other Works in the same component, so we use the "component ID" as an opaque string. That goes in a DynamoDB GSI, so you can do something like `index.findAll(componentId = …)` to find all the Works in that component.

I can't remember why exactly we did it this way, but I think the idea is we want these component IDs to be relatively stable for easier debugging across pipelines.

Does that make sense?